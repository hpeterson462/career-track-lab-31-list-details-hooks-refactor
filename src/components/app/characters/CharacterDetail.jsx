import React from 'react';
import PropTypes from 'prop-types';

const CharacterDetail = ({ name, gender, occupation, image, description }) => (
  <>
    <section>
      <p>Name{name}</p>
      <p>Gender{gender}</p>
      <p>Occupation{occupation}</p>
      <p>Image{image}</p>
      <p>Description{description}</p>
    </section>
  </>
)

CharacterDetail.propTypes = {
  name: PropTypes.string.isRequired,
  gender: PropTypes.string.isRequired,
  occupation: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
}

export default CharacterDetail;
