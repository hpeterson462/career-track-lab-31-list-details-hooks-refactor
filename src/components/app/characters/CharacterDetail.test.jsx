import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterDetail from './CharacterDetail';

describe('CharacterDetail component', () => {
  it('renders CharacterDetail after loading...', () => {
    const { asFragment } = render(<CharacterDetail
      name="Absalom"
      gender="Male"
      occupation="Cult Leader"
      image="www.x-files-image.com"
      description="Cult Lader"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
