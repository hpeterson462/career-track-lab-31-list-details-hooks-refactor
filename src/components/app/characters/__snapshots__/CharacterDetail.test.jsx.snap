// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetail component renders CharacterDetail 1`] = `
<DocumentFragment>
  <section>
    <p>
      NameAbsalom
    </p>
    <p>
      GenderMale
    </p>
    <p>
      OccupationCult Leader
    </p>
    <p>
      Imagewww.x-files-image.com
    </p>
    <p>
      DescriptionCult Lader
    </p>
  </section>
</DocumentFragment>
`;
